spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/transaction/jta/NarayanaJtaConfiguration.java

```diff
@@ -114,10 +115,12 @@ public TransactionManager narayanaTransactionManager() {
 	@Bean
 	@DependsOn("narayanaConfiguration")
 	public RecoveryManagerService narayanaRecoveryManagerService() {
+		RecoveryManager.delayRecoveryManagerThread();
 		return new RecoveryManagerService();
 	}
 
 	@Bean
+	@ConditionalOnMissingBean
```

Comment:
Can you please explain how this relates to delaying  starting the recovery manager?